lib/client/connection.js

```diff
@@ -714,3 +721,52 @@ Connection.prototype._handleSnapshotFetch = function(error, message) {
   delete this._snapshotRequests[message.id];
   snapshotRequest._handleResponse(error, message);
 };
+
+Connection.prototype.getPresence = function(channel) {
+  var connection = this;
+  return util.digOrCreate(this._presences, channel, function() {
+    return new Presence(connection, channel);
+  });
+};
+
+Connection.prototype.getDocPresence = function(collection, id) {
+  var channel = DocPresence.channel(collection, id);
+  var connection = this;
+  return util.digOrCreate(this._presences, channel, function() {
+    return new DocPresence(connection, collection, id);
+  });
+};
+
+Connection.prototype._sendPresenceAction = function(action, seq, presence) {
+  // Ensure the presence is registered so that it receives the reply message
+  this._addPresence(presence);
+  var message = {a: action, ch: presence.channel, seq: seq};
+  this.send(message);
+  return message.seq;
+};
+
+Connection.prototype._addPresence = function(presence) {
+  util.digOrCreate(this._presences, presence.channel, function() {
+    return presence;
+  });
+};
+
+Connection.prototype._handlePresenceSubscribe = function(error, message) {
+  var presence = util.dig(this._presences, message.ch);
+  if (presence) presence._handleSubscribe(error, message.seq);
+};
+
+Connection.prototype._handlePresenceUnsubscribe = function(error, message) {
+  var presence = util.dig(this._presences, message.ch);
+  if (presence) presence._handleUnsubscribe(error, message.seq);
+};
+
+Connection.prototype._handlePresence = function(error, message) {
+  var presence = util.dig(this._presences, message.ch);
+  if (presence) presence._receiveUpdate(error, message);
+};
+
+Connection.prototype._handlePresenceRequest = function(error, message) {
+  var presence = util.dig(this._presences, message.ch);
+  presence._broadcastAllLocalPresence(error, message);
```

Comment:
Is this intentionally not null-checked like the others?

Reply:
Good catch!